src/Uno.UI.Runtime.Skia.Wpf/Extensions/UI/Xaml/Controls/PasswordTextBoxView.cs

```diff
@@ -0,0 +1,157 @@
+﻿#nullable enable
+
+using System;
+using System.Windows;
+using Uno.Disposables;
+using Uno.UI.Runtime.Skia.Wpf.Controls;
+using WpfElement = System.Windows.UIElement;
+using WpfFrameworkElement = System.Windows.FrameworkElement;
+using WpfGrid = System.Windows.Controls.Grid;
+using WpfPasswordBox = System.Windows.Controls.PasswordBox;
+
+namespace Uno.UI.Runtime.Skia.Wpf.Extensions.UI.Xaml.Controls;
+
+internal class PasswordTextBoxView : WpfTextBoxView
+{
+	private readonly WpfTextViewTextBox _textBox = new();
+	private readonly WpfPasswordBox _passwordBox = new();
+	private readonly WpfGrid _grid = new();
+	private EventHandler? _textChangedWatcher;
+
+	public PasswordTextBoxView()
+	{
+		_passwordBox.BorderBrush = System.Windows.Media.Brushes.Transparent;
+		_passwordBox.Background = System.Windows.Media.Brushes.Transparent;
+		_passwordBox.BorderThickness = new Thickness(0);
+		_textBox.Visibility = Visibility.Collapsed;
+		_passwordBox.Visibility = Visibility.Collapsed;
```

Comment:
I'm don't quite understand this. We we set both to Collapsed?

If I understand correctly, the goal here is to have both a TextBox and a PasswordBox and show only one of them at a time. This is due to WPF not supporting password reveal on PasswordBox.

So, don't we always want one of them collapsed and the other visible?

Reply:
Setting both collapsed until the first `UpdateProperties` call which will display the one that matches the password reveal mode that is currently set. The method is called right after, so it works as expected